pkg/translator/faro/faro_to_logs.go

```diff
@@ -25,6 +26,46 @@ type kvTime struct {
 	hash uint64
 }
 
+var (
+	// Compile regex patterns once for performance
+	propertyAccessRegex = regexp.MustCompile(`Cannot read (property|properties) '([^']+)'`)
+	methodCallRegex     = regexp.MustCompile(`Cannot read (property|properties) '([^']+)' of`)
+	urlRegex            = regexp.MustCompile(`https?://[^\s<>"{}|\\^` + "`" + `\[\]]+`)
+	memoryAddressRegex  = regexp.MustCompile(`0x[0-9a-fA-F]+`)
+	uuidRegex           = regexp.MustCompile(`[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}`)
+	numericIDRegex      = regexp.MustCompile(`\b(id|ID|Id)\s*[:\s=]\s*\d+\b`)
+	timestampRegex      = regexp.MustCompile(`\d{4}-\d{2}-\d{2}[T ]\d{2}:\d{2}:\d{2}`)
+	filePathRegex       = regexp.MustCompile(`(?:[A-Za-z]:)?[/\\][\w\-._/\\]+\.(js|ts|jsx|tsx|css|html)\b`)
+)
+
+// drainExceptionValue normalizes exception values by replacing instance-specific
+// identifiers with placeholders to improve exception grouping by hash.
+func drainExceptionValue(value string) string {
```

Comment:
Is there an existing pattern for calling this kind of normalization draining? To me draining implies that we're removing items from a queue, not anonymizing content.

Reply:
@dehaansa i stole the term from something kinda similar - DRAIN is an algorithm to parse log lines into structure. https://jiemingzhu.github.io/pub/pjhe_icws2017.pdf

What i'm doing here is a subset of that and not reversible.